pkg/openstack/common.go

```diff
@@ -27,3 +34,119 @@ func EnsureDeleted(ctx context.Context, helper *helper.Helper, obj client.Object
 	return ctrl.Result{}, nil
 
 }
+
+// GetExternalEnpointDetailsForEndpoint - returns the MetalLBData for the endpoint, if not specified nil.
+func GetExternalEnpointDetailsForEndpoint(
+	externalEndpoints []corev1.MetalLBConfig,
+	endpt service.Endpoint,
+) *corev1.MetalLBConfig {
+	for _, metallbcfg := range externalEndpoints {
+		if metallbcfg.Endpoint == endpt {
+			return metallbcfg.DeepCopy()
+		}
+	}
+
+	return nil
+}
+
+// ServiceDetails - service details to create service.Service with overrides
+type ServiceDetails struct {
+	Name              string
+	Namespace         string
+	ServiceLabels     map[string]string
+	ServiceProtocol   k8s_corev1.Protocol
+	Endpoint          service.Endpoint
+	EndpointData      endpoint.Data
+	ExternalEndpoints []corev1.MetalLBConfig
+	OverrideSpec      []service.OverrideSpec
```

Comment:
These mighty be better represented as map[Endpoint]corev1.MetalLBConfig and map[Endpoint]service.OverrideSpec to encode that we allow one item per endpoint type